src/Entity/OgRole.php

```diff
@@ -6,177 +6,240 @@
  */
 namespace Drupal\og\Entity;
 
-use Drupal\Core\Entity\ContentEntityBase;
-use Drupal\Core\Entity\EntityTypeInterface;
-use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\og\Og;
+use Drupal\system\Tests\Common\PageRenderTest;
+use Drupal\user\Entity\User;
 
 /**
- * todo: Find a way to attach the roles to group:
- *  - UUID?
- *  - Service?
- *
- * Convert into
- *
- *
- * @ContentEntityType(
+ * @ConfigEntityType(
  *   id = "og_role",
  *   label = @Translation("OG role"),
- *   module = "og",
- *   base_table = "og_role",
+ *   static_cache = TRUE,
  *   entity_keys = {
- *     "id" = "rid",
- *     "label" = "name"
+ *     "id" = "id",
+ *     "label" = "label",
+ *     "group_type" = "groupType",
+ *     "group_bundle" = "groupBundle",
+ *     "uid" = "uid",
+ *     "permissions" = "permissions"
  *   },
+ *   config_export = {
+ *     "id",
+ *     "label",
+ *     "group_type",
+ *     "group_bundle",
+ *     "uid",
+ *     "permissions"
+ *   }
  * )
  */
-class OgRole extends ContentEntityBase {
+class OgRole extends ConfigEntityBase {
 
   /**
-   * @param mixed $gid
+   * @var integer
    *
-   * @return $this
+   * The identifier of the role.
    */
-  public function setGid($gid) {
-    $this->set('gid', $gid);
-    return $this;
-  }
+  protected $id;
+
+  /**
+   * @var string
+   *
+   * The label of the role.
+   */
+  protected $label;
+
+  /**
+   * @var integer
+   *
+   * The group ID.
+   */
+  protected $groupID;
 
   /**
-   * @return mixed
+   * @var string
+   *
+   * The group type. i.e: node, user
    */
-  public function getGid() {
-    return $this->get('gid')->value;
+  protected $groupType;
+
+  /**
+   * @var string
+   *
+   * The group bundle. i.e: article, page
+   */
+  protected $groupBundle;
+
+  /**
+   * @var integer
+   *
+   * The user ID which the role assign to.
+   */
+  protected $uid;
+
+  /**
+   * @var array
+   *
+   * List of permissions.
+   */
+  protected $permissions;
+
+  /**
+   * @return int
+   */
+  public function getId() {
```

Comment:
Why do we need this? All entities already have a id () method that is consistent.